src/components/network-chart/index.js

```diff
@@ -0,0 +1,235 @@
+        <div className="itemLabel" title={marker.data.URI}>
+          {this._shortenURI(marker.data.URI)}
+        </div>
+        <div
+          className="itemBar"
+          style={markerStyle}
+          onMouseEnter={this._hoverIn}
+          onMouseLeave={this._hoverOut}
+        >
+          &nbsp;
+        </div>
+        {this.state.hovered ? (
+          <Tooltip mouseX={this.state.pageX} mouseY={this.state.pageY}>
+            <MarkerTooltipContents
+              marker={marker}
+              threadIndex={this.props.threadIndex}
+            />
+          </Tooltip>
+        ) : null}
+      </section>
+    );
+  }
+}
+
+class NetworkChart extends React.PureComponent<Props> {
+  constructor(props: any) {
+    super(props);
+    (this: any)._renderRow = this._renderRow.bind(this);
+    (this: any)._getMarkerLength = this._getMarkerLength.bind(this);
+    (this: any)._getMarkerPosition = this._getMarkerPosition.bind(this);
+    (this: any)._getMarkerStyling = this._getMarkerStyling.bind(this);
+  }
+
+  /**
+   * Determine the maximum zoom of the viewport.
+   */
+  getMaximumZoom(): UnitIntervalOfProfileRange {
+    const { timeRange: { start, end }, interval } = this.props;
+    return interval / (end - start);
+  }
+
+  _getMarkerPosition(markerStart) {
+    const timeRange = this.props.timeRange;
+    const timeRangeTotal = timeRange.end - timeRange.start;
+
+    return (markerStart - timeRange.start) * 75 / timeRangeTotal;
+  }
+
+  _getMarkerLength(markerDuration) {
+    const timeRange = this.props.timeRange;
+    const timeRangeTotal = timeRange.end - timeRange.start;
+
+    let markerLength = markerDuration / timeRangeTotal * 75;
+
+    if (markerLength < 0.15) {
+      markerLength = 0.15;
+    }
+
+    return markerLength;
+  }
+
+  _getMarkerStyling(marker) {
+    const markerLength = this._getMarkerLength(marker.dur);
+    const markerPosition = this._getMarkerPosition(marker.data.startTime);
+
+    const markerStyling = {
+      width: markerLength + '%',
+      left: markerPosition + '%',
+    };
+    return markerStyling;
+  }
+
+  // Create row with correct details
+  _renderRow(nodeId: any, index: number) {
+    const markersPerRange = this._filterMarkersPerRange(
+      this.props.timeRange,
+      this.props.markers
+    );
+
+    return (
+      <NetworkChartRow
+        marker={markersPerRange[index]}
+        index={index}
+        markerStyle={this._getMarkerStyling(markersPerRange[index])}
+        threadIndex={this.props.threadIndex}
+      />
+    );
+  }
+
+  _filterMarkersPerRange(timeRange: Object, markers: []) {
+    return markers.filter(function(marker) {
+      const updateRange =
+        timeRange.start < marker.data.startTime &&
+        timeRange.end > marker.data.endTime;
+      return updateRange;
+    });
+  }
+
+  render() {
+    const { timeRange, markers } = this.props;
+
+    //eslint-disable-next-line
+    const networkMarkersPerRange = this._filterMarkersPerRange( timeRange, markers );
```

Comment:
This has the same performance issue I outline below: `items` will always be different (because that's what `Array.prototype.filter` does) and as a result `VirtualList` will always rerender, which is a bad thing for performance.

The solution is to make this filtering a selector, that will be memoized. Greg in #1216 introduces `getNetworkChartTracingMarkers` in `reducers/profile-view`, I think you could reuse this here.

Although I think this doesn't account for the PreviewSelection (the selection we do when we select a range in the timeline but _don't_ commit it yet) so you will likely need to change this selector to take it into account. Maybe this is as simple as using `getPreviewFilteredTracingMarkers` instead of `getSearchFilteredTracingMarkers` to define `getNetworkChartTracingMarkers`.